### DISCUSSION TITLE
Problem with Verison 1.0

### DISCUSSION THREAD

**Discussion by KnauberDirk** [dfu-programmer/dfu-programmer#85] [Q&A]
Hi. I use your DFU-Programmer for a small porject. 

I got a new PC and now, everytime i start the dfu-programmer.exe i get an Error - "0xc00007b"
So i dicided to check if there is a newer Version of the dfu-programmer and i donwloaded version 1.0.
With these Version i don't get the Error and i can start the software but i can't find my connected device anymore.
I always get "no device present".

On top - Version 1.0.0 - the lower one is Version 0.9.0

![DFU-Programmer](https://user-images.githubusercontent.com/122036050/210814112-c472e662-7b85-4025-b3b2-43b229e1a08d.png)

**Comment by cinderblock** [comment]
Can you run it with `--debug 10000`? Can you please attach the text of the output?

One change is that we switched to libusb1. That should support all the old usb drivers on Windows but we don't have a way to test them all reliably and repeatably.

**Reply by KnauberDirk** [reply]
Hi, thanks for your answer. 

I ran it with --debug 10000 and this is the output:

```log
> PS M:\ungesichert\user\dirk.knauber\dfu-driver\Neuer Ordner\dfu-programmer-x64-1.0.0> .\dfu-programmer.exe at32uc3c1512 get --debug 1000
     target: at32uc3c1512
    chip_id: 0x2feb
  vendor_id: 0x03eb
    command: get
      quiet: false
      debug: 1000
device_type: AVR32
------ command specific below ------
       name: 0
dfu.c:318: dfu_device_init( 1003, 12267, 0000001834dff910, false, false )
dfu.c:322: dfu_device_init(000003eb, 00002feb)
dfu.c:336:  0: 0x046d, 0xc050
dfu.c:336:  1: 0x0b0e, 0x0301
dfu.c:336:  2: 0x03eb, 0x2feb
dfu.c:346: found device at USB:1,4
dfu.c:502: dfu_find_interface()
dfu.c:512: config 0: MaxPower=50*2 mA
dfu.c:519: interface 0
dfu.c:526: setting 0: class:0, subclass 0, protocol:0
dfu.c:541: Found DFU Interface: 0
dfu.c:336:  3: 0x8086, 0xa2af
dfu.c:336:  4: 0x046a, 0xb090
dfu-programmer: no device present.
```

**Comment by cinderblock** [comment]
Do you have the ability to build your own version from source? (I can trigger a build here if you need.)

I'm wondering if the issue is an old bit of code relating to "`honor_interfaceclass`".

In `arguments.c`, near line `523`, can you set `args->honor_interfaceclass = true;` instead of `false` and try?

@skirridsystems Any thoughts here?

**Reply by cinderblock** [reply]
@KnauberDirk Any updates?

**Reply by KnauberDirk** [reply]
Yes i tried to build it by my own, but it doesn't work (I don't know why)
You offered to trigger a build on your side, that would help me. So i can test the changes.

**Reply by cinderblock** [reply]
See https://github.com/dfu-programmer/dfu-programmer/actions/runs/3947600618 for some custom builds for you. They're at the bottom.

**Reply by KnauberDirk** [reply]
Looks quite similar

```
PS M:\ungesichert\user\dirk.knauber\dfu-driver\Neuer Ordner\Windows x64 - dfu-programmer> .\dfu-programmer.exe at32uc3c1512 get
dfu-programmer: no device present.

PS M:\ungesichert\user\dirk.knauber\dfu-driver\Neuer Ordner\Windows x64 - dfu-programmer> 
.\dfu-programmer.exe at32uc3c1512 get --debug 1000
     target: at32uc3c1512
    chip_id: 0x2feb
  vendor_id: 0x03eb
    command: get
      quiet: false
      debug: 1000
device_type: AVR32
------ command specific below ------
       name: 0

dfu.c:317: dfu_device_init( 1003, 12267, 000000eeb21ff6d0, false, true )
dfu.c:321: dfu_device_init(000003eb, 00002feb)
dfu.c:335:  0: 0x046d, 0xc050
dfu.c:335:  1: 0x0b0e, 0x0301
dfu.c:335:  2: 0x03eb, 0x2feb
dfu.c:346: found device at USB:1,4
dfu.c:522: dfu_find_interface()
dfu.c:532: config 0: MaxPower=50*2 mA
dfu.c:539: interface 0
dfu.c:546: setting 0: class:0, subclass 0, protocol:0
dfu.c:354: Failed to find interface.
dfu.c:335:  3: 0x8086, 0xa2af
dfu.c:335:  4: 0x046a, 0xb090
dfu-programmer: no device present.

PS M:\ungesichert\user\dirk.knauber\dfu-driver\Neuer Ordner\Windows x64 - dfu-programmer> .\dfu-programmer.exe --version
dfu-programmer test-for-KnauberDirk 795def781ad12da700096d100c72c933d8e245ab
```

**Reply by cinderblock** [reply]
Hmmm. I'm not sure what is going on here. I'm new to the project and have been focusing on getting unit tests to work.

I'm having some other issues with v1.0 at the moment. Some devices seem to need to be erased before they accept any command. I'm not sure if this is related or not.

Can you plug another programmer into the device and read what flash *is* programmed currently, including the bootloader?

This *should* work.

**Comment by skirridsystems** [comment]
Sorry, that's outside of what I know about it.
Simon
